app/payment-link-v2/reference/post-index.controller.test.js

```diff
@@ -0,0 +1,220 @@
+'use strict'
+
+const proxyquire = require('proxyquire')
+const sinon = require('sinon')
+const { expect } = require('chai')
+
+const productFixtures = require('../../../test/fixtures/product.fixtures')
+const responseSpy = sinon.spy()
+const Product = require('../../models/Product.class')
+
+const mockResponses = {
+  response: responseSpy
+}
+
+let req, res
+
+describe('Reference Page - POST controller', () => {
+  const mockCookie = {
+    getSessionVariable: sinon.stub(),
+    setSessionVariable: sinon.stub()
+  }
+
+  const controller = proxyquire('./post-index.controller', {
+    '../../utils/response': mockResponses,
+    '../../utils/cookie': mockCookie
+  })
+
+  describe('when the product has no price', () => {
+    const product = new Product(productFixtures.validCreateProductResponse({
+      type: 'ADHOC',
+      reference_enabled: true,
+      price: null
+    }).getPlain())
+
+    beforeEach(() => {
+      mockCookie.getSessionVariable.reset()
+      mockCookie.setSessionVariable.reset()
+      responseSpy.resetHistory()
+    })
+
+    it('when a valid reference is entered, it should save the reference to the session and ' +
+      'redirect to the amount page', () => {
+      req = {
+        correlationId: '123',
+        product,
+        body: {
+          'payment-reference': 'valid reference'
+        }
+      }
+
+      res = {
+        redirect: sinon.spy()
+      }
+
+      controller(req, res)
+
+      expect(mockCookie.setSessionVariable.called).to.equal(true)
+      expect(mockCookie.setSessionVariable.args[0]).to.include(req)
+      expect(mockCookie.setSessionVariable.args[0]).to.include('referenceNumber')
+      expect(mockCookie.setSessionVariable.args[0]).to.include('valid reference')
+
+      expect(res.redirect.called).to.equal(true)
```

Comment:
Same here, would be nice to use the sinon assertions library

Reply:
Nice one - updated all occurrences of this.